code/modules/mining/smelter.dm

```diff
@@ -0,0 +1,184 @@
+/obj/machinery/smelter
+	name = "smelter"
+	desc = "An old Sendarian tool."
+	icon = 'icons/obj/blacksmithing.dmi'
+	icon_state = "forge"
+	density = 1
+	anchored = 0
+
+/obj/machinery/smelter/attackby(obj/item/weapon/W, mob/user, params)
+	var/smelting_result = W.on_smelt()
+	if(!smelting_result)
+		return ..()
+	if(user.drop_item())
+		to_chat(user, "You smelt [W].")
+		qdel(W)
+		var/obj/item/weapon/reagent_containers/glass/bucket/AB = new(get_turf(src))
+		AB.reagents.add_reagent(smelting_result, 75)
+		AB.reagents.chem_temp = 1000
+		AB.reagents.handle_reactions()
+		AB.name = "bucket of [AB.reagents.get_master_reagent_name()]"
+
+
+/obj/machinery/anvil
+	name = "anvil"
+	desc = "Goodman Durnik, is that you?"
+	icon = 'icons/obj/blacksmithing.dmi'
+	icon_state = "anvil"
+	density = 1
+	anchored = 0
+	var/obj/item/weapon/reagent_containers/glass/mold/current_mold = null
+	var/mutable_appearance/my_mold = null
+
+/obj/machinery/anvil/attackby(obj/item/weapon/W, mob/user, params)
+	if(!current_mold && istype(W, /obj/item/weapon/reagent_containers/glass/mold))
+		var/obj/item/weapon/reagent_containers/glass/mold/M = W
+		var/datum/reagent/R = M.reagents.get_master_reagent()
+		if(R && R.volume == 25)
```

Comment:
These 3 checks are kinda bad. Since if someone tries a mold with a volume bigger than 25 it will fail. This whole thing could be:
```
if(!R || R.volume < 25)
    not enough
    return
do the placement
```